### DISCUSSION TITLE
📁 Reordering `migrations` Folder

### DISCUSSION THREAD

**Discussion by taljacob2** [redhat-beyond/JobSeeker#56] [General]
@DeanBiton @paOmer @amittcohen @ShirleyFichman

# 📁 Reordering `migrations` Folder

## Description

Hi all,
I have noticed that if you `makemigrations` multiple times, without deleting the old migrations you already had,
then *Django* will create for you another file that *alters* the old migration file.

See screenshot example:
![image](https://user-images.githubusercontent.com/70590583/160991132-4d692ae3-0147-4c72-947e-57aaed9729e3.png)

So this could lead the `migrations` folder having some **disadvantages:**
- Redundant files / size / rendering time.
- Misleading the final result of the migration when reviewing the files.

## Solution

1. In the `migrations` folder, **delete** the files:
    - `0001_initial.py`
    - Files that  has `alter` in their name.

1. Run `makemigrations` again.

### Result
All those migration files you have deleted above will be updated to a new single `0001_initial.py` file.

**Comment by taljacob2** [comment]
@paOmer This will also resolve our discussion we had here: https://github.com/redhat-beyond/JobSeeker/pull/27#issuecomment-1082965698

**Comment by ifireball** [comment]
You should not do this - you need the older migration files to track the status changes of your database - the only time you can clean up migrations is *before you merge them* - for example you can make sure you only add a single file for the set of changes in a given PR.

**Reply by taljacob2** [reply]
@ifireball 
Thanks for review.
Okay, so you mean that once the migrations had been merged to the `main`, we will not use this "delete and re-`makemigrations`" solution.
But in case there is a pending PR that had not been merged yet, should we use this solution?

### EDITED:
> But in case there is a pending PR that had not been merged yet, should we use this solution?

But in case there is a pending PR that had not been merged yet, is it **required** to use this solution?

**Reply by ifireball** [reply]
Not required, but useful to make the PR simpler. As with everything else, the point of what you out in the PR is to make it as easy as possible for the reader to understand what you did and why.
